src/sidebar/components/test/group-list-v2-test.js

```diff
@@ -0,0 +1,166 @@
+'use strict';
+
+const { shallow } = require('enzyme');
+const { createElement } = require('preact');
+
+const GroupList = require('../group-list-v2');
+
+describe('GroupList', () => {
+  let fakeServiceConfig;
+  let fakeServiceUrl;
+  let fakeSettings;
+  let fakeStore;
+
+  const testGroup = {
+    id: 'testgroup',
+    name: 'Test group',
+    organization: { id: 'testorg', name: 'Test Org' },
+  };
+
+  function createGroupList() {
+    return shallow(
+      <GroupList serviceUrl={fakeServiceUrl} settings={fakeSettings} />
+    ).dive();
+  }
+
+  beforeEach(() => {
+    fakeServiceUrl = sinon.stub();
+    fakeSettings = {
+      authDomain: 'hypothes.is',
+    };
+    fakeStore = {
+      getCurrentlyViewingGroups: sinon.stub().returns([]),
+      getFeaturedGroups: sinon.stub().returns([]),
+      getMyGroups: sinon.stub().returns([]),
+      focusedGroup: sinon.stub().returns(testGroup),
+      profile: sinon.stub().returns({ userid: null }),
+    };
+    fakeServiceConfig = sinon.stub().returns(null);
+
+    GroupList.$imports.$mock({
+      '../store/use-store': callback => callback(fakeStore),
+      '../service-config': fakeServiceConfig,
+    });
+  });
+
+  afterEach(() => {
+    GroupList.$imports.$restore();
+  });
+
+  it('displays no sections if there are no groups', () => {
+    const wrapper = createGroupList();
+    assert.isFalse(wrapper.exists('GroupListSection'));
+  });
+
+  it('displays "Currently Viewing" section if there are currently viewing groups', () => {
+    fakeStore.getCurrentlyViewingGroups.returns([testGroup]);
+    const wrapper = createGroupList();
+    assert.isTrue(
+      wrapper.exists('GroupListSection[heading="Currently Viewing"]')
+    );
+  });
+
+  it('displays "Featured Groups" section if there are featured groups', () => {
+    fakeStore.getFeaturedGroups.returns([testGroup]);
+    const wrapper = createGroupList();
+    assert.isTrue(
+      wrapper.exists('GroupListSection[heading="Featured Groups"]')
+    );
+  });
+
+  it('displays "My Groups" section if user is a member of any groups', () => {
+    fakeStore.getMyGroups.returns([testGroup]);
+    const wrapper = createGroupList();
+    assert.isTrue(wrapper.exists('GroupListSection[heading="My Groups"]'));
+  });
+
+  it('sorts groups within each section by organization', () => {
```

Comment:
I'm having a little trouble discerning exactly what this is testing...I'm wondering if the term "sort" in the test description is misleading. Is this test checking to make sure that groups are _slotted_ into the appropriate section?

Reply:
This is about making sure that the order of groups within a section matches what `groupOrganizations(groupsForSection)` returns rather than say, being random or whatever came back from the server.